### PR TITLE
Add rewardful extension

### PR DESCRIPTION
**Pull request by AlexIsMaking**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/rewardful/src/types.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
+// General types used throughout the app
+export type Preferences = {
+  apiKey: string;
+};
```
Comment (pernielsentikaer) [inline extensions/rewardful/src/types.ts:4]:
```suggestion
```
This is already included in `raycast-env.d.ts`

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have a single feedback ready for you 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by AlexIsMaking** [comment]
@pernielsentikaer thanks for the heads up, could you please point me to the docs which explain how I should enable the built-in API key storage preference so that users can submit theirs? Searching the docs for `raycast-env.d.ts` isn't showing any relevant results and I can't see a mention of this in the docs for preferences either.

**Comment by pernielsentikaer** [comment]
Are you okay with the corrections here @AlexIsMaking

**Comment by AlexIsMaking** [comment]
@pernielsentikaer those are fine by me thanks. I went for "Affiliates", "Commissions" etc. because you can do more than just view those items (e.g. create new affiliates) but I'm happy to use whichever naming convention you prefer.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**